db/commitlog/commitlog.cc

```diff
@@ -904,41 +946,77 @@ class db::commitlog::segment : public enable_shared_from_this<segment>, public c
         }
 
         if (_closed) {
-            return make_exception_future<rp_handle>(std::runtime_error("commitlog: Cannot add data to a closed segment"));
+            return make_exception_future<>(std::runtime_error("commitlog: Cannot add data to a closed segment"));
         }
 
         buf_memory -= permit.release();
         _segment_manager->account_memory_usage(buf_memory);
 
-        replay_position rp(_desc.id, position());
-        _cf_dirty[id]++; // increase use count for cf.
+        auto& out = _buffer_ostream;
+
+        std::optional<crc32_nbo> mecrc;
+
+        // if this is multi-entry write, we need to add an extra header + crc
+        // the header and crc formula is:
+        // header:
+        //      magic : uint32_t
```

Comment:
May be worth adding `uint32_t flags` field for future extensions and check that it is zero while reading.

Reply:
Can just bump the version again if any more info is needed, no?

Reply:
Yeah. The code that parses it will start to be more and more cumbersome though. But it is possible. Flags make it easier to backport features selectively, not sure it is needed here.

Reply:
I am opting against. If the format needs changing, I'll bump again. Saving space.

Reply:
I will not insist, though feature flags are superior to versioning  IMO.